### PR TITLE
The common code for the partial restore

### PR DESCRIPTION
**Pull request by andr-sokolov**
Add the --filter option for the archive-get and restore commands and help on it.
Add the isRelationNeeded function to check whether relation should be restored.
The function makes decision on the contents of JSON file from the --filter option.

Co-authored-by: Andrey Sokolov <a.sokolov@arenadata.io>

### CODE REVIEW

File: test/src/module/common/partialRestoreTest.c [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        HRN_CFG_LOAD(cfgCmdRestore, strLstDup(argListCommon));
+        TEST_RESULT_BOOL(isRelationNeeded(16384, 1663, 16390), true, "always true without --filter");
+
+        TEST_TITLE("The --filter option is invalid");
+        StringList *const argListInv = strLstDup(argListCommon);
+        hrnCfgArgRawZ(argListInv, cfgOptFilter, "recovery_filter.json");
+        HRN_CFG_LOAD(cfgCmdRestore, argListInv);
+        TEST_ERROR(isRelationNeeded(16384, 1663, 16390), AssertError, "The path to the filter info file is not absolute");
+
+        TEST_TITLE("The --filter option is corrent");
```
Comment (KnightMurloc) [inline test/src/module/common/partialRestoreTest.c]:
```suggestion
        TEST_TITLE("The --filter option is correct");
```

File: src/common/partialRestore.c [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        jsonReadFree(jsonRead);
+        bufFree(jsonFile);
+        storageReadFree(storageRead);
+    }
+
+    const DataBase *const db = lstFind(filterList, &dbNode);
+    if (db == NULL)
+        return false;
+
+    return pgDbIsSystemId(relNode) || lstExists(db->tables, &(Table){.spcNode = spcNode, .relNode = relNode});
```
Comment (KnightMurloc) [inline src/common/partialRestore.c]: Line is too long.

File: src/common/partialRestore.c [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    List *tables;
+} DataBase;
+
+typedef struct Table
+{
+    Oid spcNode;
+    Oid relNode;
+} Table;
+
+static List *filterList = NULL;
```
Comment (KnightMurloc) [inline src/common/partialRestore.c]: Maybe make this variable static inside a function? It should not be used outside of it anyway.

File: test/src/module/common/partialRestoreTest.c (hunk 1 above)
Comment (andr-sokolov) [reply test/src/module/common/partialRestoreTest.c]: Fixed

File: src/common/partialRestore.c (hunk 2 above)
Comment (andr-sokolov) [reply src/common/partialRestore.c]: Fixed

File: src/common/partialRestore.c (hunk 3 above)
Comment (andr-sokolov) [reply src/common/partialRestore.c]: Done

**Review by KnightMurloc — APPROVED** [review]

**Pull request merged**